MON-157180-v3-gorgone-cloud-poller-disconnect-when-gorgone-log-are-to-big-to-be-sent (#2571)
centreon/centreon-collect: gorgone/docs/modules/core/proxy.md

    \"cache_dir\": \"/var/cache/centreon\"
}"
```

### Developer manual

> This module use register and node module to get the list of node to manage.

Reviewer: ```suggestion
This module uses **register** and **node modules** to get the list of nodes to manage.

```

Author: ```suggestion
This module uses **register** and **node** modules to get the list of nodes to manage.

```

> This module use multiples process : one control, a pool of worker to process event and optionally an httpserver process if pullwss is used.

Reviewer: ```suggestion
This module uses multiple processes: one controls a pool of workers to process events and optionally an httpserver process if pullwss is used.

```

## check()

> Run by gorgone-core process regularly (5s).

Reviewer: ```suggestion
Run by the gorgone-core process regularly (5s).

```

> - start an history synchronization if needed

Reviewer: ```suggestion
- start a history synchronization if needed

```

- delete old history synchronization if older than synchistory_timeout
- launch a ping to all nodes

> - delete old ping if older than pong_discard_timeout

Reviewer: ```suggestion
- delete old pings if older than ping_discard_timeout

```

Author: ```suggestion
- delete old pings if older than **pong_discard_timeout**

```

> for the sync history, the process is as follows : [sequence diagram](./pullwss-log-sync.mmd)

Reviewer: ```suggestion
For the sync history, the process is as follows : [sequence diagram](./pullwss-log-sync.mmd)

```

> 2 thing can start an history sync, synchistory_time configuration, run by check()

Reviewer: ```suggestion
Two things can start a history sync:

* synchistory_time configuration, run by check()

```

> and an api call to a specific node (in the form [/api/nodes/:nodeid/...](../../api.md)). It is not shown on the above sequence diagram.

Reviewer: ```suggestion
* an api call to a specific node (in the form [/api/nodes/:nodeid/...](../../api.md)). It is not shown on the above sequence diagram.

```
